### PR TITLE
Add enrollment code info

### PR DESCRIPTION
**Pull request by srpearce**

### CODE REVIEW

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 1]
```diff
@@ -1 +1,223 @@
...
+#. At the bottom of the page, select **Save**.
+
+
+Assign Admin Roles
+************************************
+
+Two new roles are required for courses that create and manage enrollment codes.
+
+* The *Sales Admin* role both creates and manages enrollment codes.
+* The *Finance Admin* role manages enrollment codes, but does not create them.
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: I'm not sure the above is correct. I know that you can generate redeem codes with sales_admin and that you can view reports with finance_admin, but I don't know if you can do both with either role.

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 2]
```diff
@@ -1 +1,237 @@
...
+
+Assign Admin Roles
+************************************
+
+Two new roles are required for courses that create and manage enrollment codes.
+
+* The *Finance Admin* role creates and manages enrollment codes and discounts.
+  This role can also download reports.
+* The *Sales Admin* role creates and manages enrollment codes, but cannot see
+  reports.
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: @chrisndodge  https://github.com/edx/edx-platform/wiki/Shopping-Cart says that the
sales_admin role can "create invoices, but [does] not have access to other
areas of the page. However, when I tested this in the sandbox, the SA role
could see the "Registration Codes", "Course Price", and "Coupons List"
areas - the only area that FA but not SA had was "Coupons List". Is this
correct?

Also, under "Coupons List," there's a button to download a .csv file of all
coupon codes for the course, but I don't see how to generate coupon codes
in the first place. How does the admin do this?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 3]
```diff
@@ -1 +1,237 @@
...
+information about your course or the transaction.
+
+.. _Download Enrollment Code Reports:
+
+=======================================
+Download Enrollment Code Reports
+=======================================
+
+To help you keep track of the enrollment codes that have been issued for your
+course, you can download .csv files that contain enrollment code reports. The reports contain the following information.
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: @chrisndodge I made up the following list; can you correct it as necessary?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 4]
```diff
@@ -1 +1,237 @@
...
+* Download a .csv file of all credit card purchases. To do this, select
+  **Download Credit Card Purchases**.
+* Cancel an invoice. To do this, enter the invoice number in the **Invoice
+  Number** field, and then select **Cancel Invoice**. To find an invoice
+  number, see the .csv file that the system creates and downloads to your
+  computer when you create enrollment codes. 
+* Resubmit an invoice. To do this, enter the invoice number in the **Invoice
+  Number** field, and then select **Resubmit Invoice**. To find an invoice
+  number, see the .csv file that the system creates and downloads to your
+  computer when you create enrollment codes.
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: I'm not sure what happens when a user cancels or resubmits an invoice, but it looks like the work from SOL-916 may supersede these buttons anyway. Is that true?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 2 above)
Comment (chrisndodge) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: A finance_admin will be able to create coupons.

Our team did not create the "sale_admin" role, so the scope of permissions might be incorrectly written up. I believe that addition came from the DedX team.

**Comment by srpearce** [comment]
@chrisndodge This is ready for review. In addition to the inline questions, I have two more:
- How does the admin override the default e-mail? The Custom Theming (https://github.com/edx/edx-platform/wiki/Custom-Theming) page doesn't cover this and I'm not sure where else to look.
- Do we need to document the "Course Price" and "Coupons List" sections of the E-Commerce page as well? 

Thanks!

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 3 above)
Comment (chrisndodge) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: Here's the list of columns in that report:
- the code itself
- the URL to redeem the code
- course id
- company_name (from the invoice)
- create_by (who created the invoice)
- reedemed_by (who redeemed the code)
- invoice_id
- email of purchaser
- customer reference (can be input into Invoice form)
- internal reference (can be input into the Invoice form)
- is_active (if the code is still active)

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 2 above)
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: @wedaly Can you shed some light on the sales_admin vs. finance_admin roles?

@chrisndodge Did you see the second question above (re: how to generate coupon codes - I don't see a way to do this in the UI)?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 4 above)
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: @chrisndodge ^^

**Comment by srpearce** [comment]
@chrisndodge Did you see the two questions above (re: overriding default e-mail and documenting the "Course Price" and "Coupons List" sections)?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 2 above)
Comment (chrisndodge) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: Can you verify that you have finance_admin rights? If you tell me which course you are looking at (and which environment), I can make sure that all permissions are granted to you.

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 4 above)
Comment (chrisndodge) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: Hrm. SOL-916 definately overlaps, but is more fine grained where one can query, enable/disable, unredeem a code.

When an invoice is invalidated , users that enrolled in a course with that registration code will be blocked from entering the course until the invoice is marked as valid again. Does that make sense?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 5]
```diff
@@ -1 +1,237 @@
...
+* A profile ID.
+* A purchase endpoint URL. 
+* A secret key.
+
+Add Code to .json Files
+************************************
+
+To enable users to pay for your course, you must add the following configuration in your Open edX environment.
+
+* Add the following code to the lms.env.json file.
```
Comment (mhoeber) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: where to you find this file?

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst [hunk 6]
```diff
@@ -1 +1,237 @@
...
+    ...
+    "FEATURES": {
+    ...
+    "ENABLE_SHOPPING_CART": true,
+    "ENABLE_PAID_COURSE_REGISTRATION": true,
+    ...
+    }
+    ....
+
+* Add the following code to the lms.auth.json file.
```
Comment (mhoeber) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: file location?

**Comment by chrisndodge** [comment]
@srpearce is this documentation expected as a deliverable for our June 8th delivery to our partner? If so, then I'd recommend yes, as those could be features they would like to use.

Re: substituting emails via theming, I look a quick look and I'm not sure how to do it (I know how to do it in White Label/Microsites). I'll keep looking and get back to you.

File: en_us/open_edx_course_authors/source/running_course/course_enrollment.rst (hunk 4 above)
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/course_enrollment.rst]: Oh gracious. I was equating enrollment codes (individual and collective) with invoices (which may contain multiple enrollment codes) in my head. [blushes] As you were...

File: en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+
+.. contents:: 
+   :depth: 1
+   :local:
+
+.. _Configure the Course:
+
+****************************
+Configure the Course
+****************************
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst]: @chrisndodge You can skip lines 24-199.

File: en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+   * - Coupons List
+     - Yes
+     - Yes
+
+
+.. _Set Course Price:
+
+******************************************
+Set the Price of the Course
+******************************************
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst]: @chrisndodge Please review lines 200-292.

File: en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst [hunk 9]
```diff
@@ -0,0 +1,475 @@
...
+#. In the **Coupon Information** list, locate the coupon code that you want to
+   edit or delete.
+#. To edit the coupon code, select **Edit** in the **Actions** column, edit
+   the description of the coupon code in the **Update Coupon** dialog box, and
+   then select **Update Coupon**.
+
+   To delete the coupon code, select the ``[x]`` in the **Actions** column.
+   The coupon code moves to the end of the list and appears crossed out with a
+   light red background.
+
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst]: @chrisndodge You can skip the rest of this file.

**Comment by srpearce** [comment]
@chrisndodge The new text is ready for review. **You only need to review lines 1-24 and 200-292**; some of the rest has been reorganized, but directions, etc. haven't changed.

File: en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst [hunk 10]
```diff
@@ -0,0 +1,475 @@
...
+#. In the **Add Coupon** dialog box, enter your information in the **Code**
+   and **Percentage Discount** fields. 
+#. (optional) In the **Description** field, enter a description for your
+   coupon code.
+#. (optional) Select **Expiration Date**, and then select a date from the
+   calendar that opens.
+#. Select **Add Coupon**.
+
+.. To do: What time does the coupon code expire on that date? Tested 00:00:00
+.. UTC and it didn't expire then, but it does expire before midnight.
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst]: @chrisndodge Can you answer the question in lines 267-8?

File: en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst [hunk 11]
```diff
@@ -0,0 +1,475 @@
...
+            }
+        }, 
+        "CC_PROCESSOR_NAME": "CyberSource2",
+    ... 
+
+.. _Add Course Mode:
+
+=============================
+Add a Course Mode in Django
+=============================
```
Comment (srpearce) [inline en_us/open_edx_course_authors/source/running_course/manage_paid_courses.rst]: @catong Instructions for adding the honor code mode, but I don't know if these steps would work for verified certificates.

**Pull request merged**